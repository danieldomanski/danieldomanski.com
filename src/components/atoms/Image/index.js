import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Img from 'gatsby-image/withIEPolyfill'

const BgImage = styled(Img)`
  margin: 0;
  transition: 0.75s ease-in-out;
  width: ${props => (props.width ? props.width : 'initial')};
`

const Image = ({ input, type, fit, ...rest }) => {
  const img =
    type === 'fluid' ? input.childImageSharp.fluid : input.childImageSharp.fixed

  return type === 'fluid' ? (
    <BgImage objectFit={fit} fluid={img} {...rest} />
  ) : (
    <BgImage objectFit={fit} fixed={img} {...rest} />
  )
}

export default Image

Image.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]).isRequired,
  type: PropTypes.oneOf(['fixed', 'fluid']),
  fit: PropTypes.string,
}

Image.defaultProps = {
  fit: 'cover',
  type: 'fluid',
}
