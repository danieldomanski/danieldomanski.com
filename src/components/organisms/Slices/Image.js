import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Img from 'gatsby-image'

const Wrapper = styled.div`
  margin-bottom: 2rem;
`

const Image = ({ data }) => (
  <Wrapper>
    <Img fluid={data.primary.image.localFile.childImageSharp.fluid} />
  </Wrapper>
)

export default Image

Image.propTypes = {
  data: PropTypes.objectOf(PropTypes.object).isRequired,
}
