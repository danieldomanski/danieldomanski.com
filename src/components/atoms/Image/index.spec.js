import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, cleanup } from '@testing-library/react'

import Image from './index'

afterEach(cleanup)

test('<Image> renders', () => {
  let mockData = {
    input: {
      childImageSharp: {
        fluid: {
          src: 'mock.png',
          aspectRatio: 0,
          sizes: '',
          srcSet: '',
          srcSetWebp: '',
          srcWebp: '',
          base64: '',
        },
      },
    },
    type: 'fluid',
  }

  const { container } = render(<Image {...mockData} />)

  expect(container.firstChild).toMatchSnapshot()
})
