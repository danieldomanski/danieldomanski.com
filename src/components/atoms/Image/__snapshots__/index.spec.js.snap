// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image> renders 1`] = `
.c0 {
  -webkit-transition: 0.75s ease-in-out;
  transition: 0.75s ease-in-out;
  max-height: px;
}

<div
  class="c0 gatsby-image-wrapper"
  style="position: relative; overflow: hidden;"
>
  <div
    style="width: 100%;"
  />
  
  <picture>
    
    <source
      sizes=""
      srcset=""
    />
    <img
      alt=""
      loading="lazy"
      sizes=""
      src="mock.png"
      srcset=""
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: 50% 50%; opacity: 0; transition: opacity 500ms; font-family: \\"object-fit: cover; object-position: 50% 50%\\";"
    />
  </picture>
  <noscript>
    &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="mock.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
  </noscript>
</div>
`;
